### PR TITLE
Fix: Return None in finite predicate Mul expression when a finite argument’s nonzero status is indeterminate

### LINKED ISSUE #27447: ask(Q.finite(f * i)) wrongly returns False (for finite f and infinite i)
```python
>>>f = Symbol('f', finite=True)
>>>i = Symbol('i', infinite=True)
>>>ask(Q.finite(f * i))
False
```

~~An infinite number times zero is zero as far as I'm aware. So since zero is a finite number, this behavior is incorrect.~~
An infinite number times zero is undefined, so the above behavior is incorrect.

### PR DESCRIPTION
**Pull request by Jatinbhardwaj-093**
#### Brief description of what is fixed or changed
Previously, in a Mul expression, if any argument was unbounded (infinite), the result would immediately be False, ignoring cases where a finite argument could be zero. However, if the finite argument is zero, the correct result should be None (undefined).

This PR accounts for that case and ensures the correct behavior by returning None when the finite argument’s value cannot be determined as nonzero.

#### Release Notes

* assumptions
  * Make Q.finite(Mul(...)) return `None` when a finite argument’s nonzero status is indeterminate instead of `False`.

### CODE REVIEW

**Comment by krishnavbajoria02** [comment]
@Jatinbhardwaj-093 I'll suggest you to edit the .mailmap file by adding your name and email id. You can refer to this link : https://docs.sympy.org/latest/contributing/new-contributors-guide/workflow-process.html#mailmap-instructions

**Comment by TiloRC** [comment]
See [this](https://docs.sympy.org/latest/contributing/new-contributors-guide/writing-tests.html#writing-tests-regression-tests) page for how to write regression tests properly.

**Comment by Jatinbhardwaj-093** [comment]
I am a little lost on what exactly I need to do since the previous test cases already check all the cases indirectly. So, should I add test cases with more than three expressions, or should I write tests that check directly without using assumptions, like this:
```python
def test_issue_27447():
    x = Symbol('x', finite=True)  
    y = Symbol('y', finite=True, nonzero=True)  
    assert ask(finite(Mul(x, oo))) is None  
    assert ask(finite(Mul(y, oo))) is False  
    assert ask(finite(Mul(oo, oo))) is False  
    assert ask(finite(Mul(0, oo))) is None 
```

Or am I completely wrong on both parts? Can you guide me further?

**Comment by TiloRC** [comment]
> I am a little lost on what exactly I need to do since the previous test cases already check all the cases indirectly.

Oh sorry, I didn't look at the code carefully and got confused. My bad.

File: sympy/assumptions/tests/test_query.py [hunk 1]
```diff
@@ -1019,9 +1019,9 @@ def test_bounded():
     a = x*y
     x, y = a.args
     assert ask(Q.finite(a), Q.finite(x) & Q.finite(y)) is True
-    assert ask(Q.finite(a), Q.finite(x) & ~Q.finite(y)) is False
+    assert ask(Q.finite(a), Q.finite(x) & ~Q.finite(y)) is None
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:1022]: Although returning None is always technically correct, we don't want to remove functionality. False is correct here.

**Comment by TiloRC** [comment]
Oops. Ignore that comment.

**Comment by TiloRC** [comment]
Ok, I know how to proceed. 

I think you should change all of the `~Q.finite(...)` to `Q.infinite(...)`. For example, `ask(Q.finite(a), Q.finite(x) & ~Q.finite(y)) is False` can become  `ask(Q.finite(a), Q.finite(x) & Q.infinite(y)) is False` and so on. 

And you can add a regression test as normal.

**Comment by Jatinbhardwaj-093** [comment]
should i change the `~Q.finite(...)` to `Q.infinite(...)`. Everywhere or only to my test scope.

File: sympy/assumptions/tests/test_query.py [hunk 2]
```diff
@@ -657,31 +658,31 @@ def test_bounded():
     assert ask(Q.finite(a), Q.finite(x) & ~Q.positive(x)
         & ~Q.positive(y)) is None
     # U + U
-    assert ask(Q.finite(a), ~Q.finite(x) & ~Q.finite(y)) is None
+    assert ask(Q.finite(a), Q.infinite(x) & Q.infinite(y)) is None
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:661]: I'm confused as to why this test is being changed.

Comment (TiloRC) [reply sympy/assumptions/tests/test_query.py:661]: I did not mean change ~Q.finite(y) -> Q.infinite(y) everywhere — just for the tests that were asserting wrong behavior.

File: sympy/assumptions/tests/test_query.py [hunk 3]
```diff
@@ -1111,6 +1112,14 @@ def test_issue_27441():
...
     assert ask(Q.composite(y), Q.integer(y) & Q.positive(y) & ~Q.prime(y)) is None
 
+def test_issue_27447():
+    x = Symbol('x', finite=True)
+    y = Symbol('y', finite=True, nonzero=True)
+    # https://github.com/sympy/sympy/issues/27447
+    assert ask(Q.finite(Mul(x, oo))) is None
+    assert ask(Q.finite(Mul(y, oo))) is False
+    assert ask(Q.finite(Mul(oo, oo))) is False
+    assert ask(Q.finite(Mul(0, oo))) is None
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:1122]: These two are not checking what you think they're checking. `Mul(oo, oo)` simplifies to oo and `Mul(0, oo)` simplifies to NaN. So you're actually checking  `ask(Q.finite(oo)) is False` and `ask(Q.finite(nan))` is None.

**Review by TiloRC — CHANGES_REQUESTED** [review]

File: sympy/assumptions/tests/test_query.py (hunk 2 above)
Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:661]: Sorry, I guess you suggested changes everywhere for uniformity. I will revert the changes.

File: sympy/assumptions/tests/test_query.py (hunk 3 above)
Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1122]: You’re right, I’ll remove these test cases.

File: sympy/assumptions/handlers/calculus.py [hunk 4]
```diff
@@ -151,10 +151,12 @@ def _(expr, assumptions):
         * /s = not signed
     """
     result = True
+    possible_zero = False
     for arg in expr.args:
         _bounded = ask(Q.finite(arg), assumptions)
         if _bounded:
-            continue
+            if ask(Q.eq(arg, 0), assumptions) is None:
```
Comment (TiloRC) [inline sympy/assumptions/handlers/calculus.py:158]: It would be better to use `Q.zero` instead of `Q.eq` here as `ask` is better at reasoning about the former.

File: sympy/assumptions/tests/test_query.py [hunk 5]
```diff
@@ -1019,9 +1020,9 @@ def test_bounded():
     a = x*y
     x, y = a.args
     assert ask(Q.finite(a), Q.finite(x) & Q.finite(y)) is True
-    assert ask(Q.finite(a), Q.finite(x) & ~Q.finite(y)) is False
+    assert ask(Q.finite(a), Q.finite(x) & Q.infinite(y)) is None
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:1023]: Would it be possible to improve the code so that `assert ask(Q.finite(a), Q.finite(x) & ~Q.finite(y)) is False` and all of the other similar assert statements passed?

File: sympy/assumptions/tests/test_query.py [hunk 6]
```diff
@@ -1111,6 +1112,12 @@ def test_issue_27441():
     # https://github.com/sympy/sympy/issues/27441
     assert ask(Q.composite(y), Q.integer(y) & Q.positive(y) & ~Q.prime(y)) is None
 
+def test_issue_27447():
+    x = Symbol('x', finite=True)
+    y = Symbol('y', finite=True, nonzero=True)
+    # https://github.com/sympy/sympy/issues/27447
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:1118]: This is not needed as the issue number is in `def test_issue_27447()`

File: sympy/assumptions/tests/test_query.py [hunk 7]
```diff
@@ -1111,6 +1112,12 @@ def test_issue_27441():
     # https://github.com/sympy/sympy/issues/27441
     assert ask(Q.composite(y), Q.integer(y) & Q.positive(y) & ~Q.prime(y)) is None
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:1114]: Match the style of the other tests. There needs to be two spaces between each test function.

**Review by TiloRC — CHANGES_REQUESTED** [review]

File: sympy/assumptions/tests/test_query.py (hunk 5 above)
Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1023]: I don’t know if I have the right context, but ~Q.finite(y) can be either infinite or undetermined. So, shouldn’t the result be None in both cases?

Comment (TiloRC) [reply sympy/assumptions/tests/test_query.py:1023]: Sorry, I wasn't clear. I meant:

Can we replace all of the incorrect assert statements like `assert ask(Q.finite(a), Q.finite(x) & ~Q.finite(y)) is False` with similar but correct assert statements like `assert ask(Q.finite(a), Q.finite(x) & Q.infinite(y)) is False` that assert False or True instead of None. And improve the code so that those statements pass.

Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1023]: However, making this change would defeat the purpose of this PR. The primary goal of this PR is to address whether 0 * oo should return NaN. When an expression contains a mix of finite and infinite arguments, and we cannot determine whether the finite argument is nonzero, we should consistently return None.

Additionally, in this part of the code, we always receive False, Q.infinite, and ~Q.finite cases:
https://github.com/sympy/sympy/blob/a8eea83ecd1591d5e4db818585304f3f60914c59/sympy/assumptions/handlers/calculus.py#L155

Thus, returning None depends on whether the finite argument could be zero. Previously, None was only returned when the expression contained an unknown argument. Now, it is also returned when we cannot determine if a finite argument is nonzero.

Comment (TiloRC) [reply sympy/assumptions/tests/test_query.py:1023]: > However, making this change would defeat the purpose of this PR

You are correct. Sorry about that and thank you for your patience. 

I do think the idea of replacing it with a statement that still returns True or False is a good one. Perhaps we could replace 
- `assert ask(Q.finite(a), Q.finite(x) & ~Q.finite(y)) is False`
with
- `assert ask(Q.finite(a), Q.finite(x) & ~Q.finite(y) & ~Q.zero(y)) is False`

Currently the old assumptions are able to make this sort of inference and we should strive to make the new assumptions at least as powerful as the old assumptions:
```python
from sympy import symbols
x = symbols('x',finite=True, nonzero=True)
y = symbols('y',infinite=True)
(x*y).is_finite # returns False
```

File: sympy/assumptions/tests/test_query.py [hunk 8]
```diff
@@ -1112,6 +1113,13 @@ def test_issue_27441():
...
 
 
+def test_issue_27447():
+    x = Symbol('x', finite=True)
+    y = Symbol('y', finite=True, nonzero=True)
+    assert ask(Q.finite(Mul(x, oo))) is None
+    assert ask(Q.finite(Mul(y, oo))) is False
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:1120]: Could you avoid using the old assumptions in these tests? In other words, define x and y like `x, y = symbols('x y')` and pass assumptions about them using `Q` to `ask` queries.

**Comment by TiloRC** [comment]
I think your release notes entry can be improved. See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for advice in writing release note entries. Remember — the release note entries are meant for end users of SymPy so they should be easily understandable by such users. Without knowing much about SymPy' assumption system, would you have understood anything from reading your current release notes entry?

**Review by TiloRC — CHANGES_REQUESTED** [review]

File: sympy/assumptions/tests/test_query.py (hunk 5 above)
Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1023]: Please correct me if I’m mistaken, but are you suggesting that it should return None only when there is a guarantee of a zero argument with infinity?

Additionally, in this expression:
assert ask(Q.finite(a), Q.finite(x) & ~Q.finite(y) & ~Q.zero(y)) is False
I was wondering about the inclusion of ~Q.zero(y). Since y is not finite, wouldn’t this condition have no effect? Perhaps you intended it for x instead?

Comment (TiloRC) [reply sympy/assumptions/tests/test_query.py:1023]: > Perhaps you intended it for x instead?
Yes, you're right — that is what I intended. 

> Please correct me if I’m mistaken, but are you suggesting that it should return None only when there is a guarantee of a zero argument with infinity?

No. If it's possible for a=0*oo=undefined, `ask(Q.anything(a))` should return None for all predicates. However, if a must be finite or infinite, we should try to return True/False, especially if the old assumptions can handle the query.

Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1022]: I have added the appropriate test case for this scenario in a separate test I created.

Now, it behaves the same way as the old assumptions system.

Regarding your suggestion for replacing the test in the test_basis function, would you recommend making the replacement, or should I keep it as a separate test?

Comment (TiloRC) [reply sympy/assumptions/tests/test_query.py:1022]: Could you make this test the following?

`assert ask(Q.finite(a), Q.finite(x) & ~Q.zero(x) & ~Q.finite(y)) is False`

> Would you recommend replacing the test in the test_basis function, or should I keep it as a separate test?

I'm not sure what the test_basis function is.

Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1022]: > I'm not sure what the test_basis function is.

Sorry, I may mention this by checking different files.

> Could you make this test the following?
>      `assert ask(Q.finite(a), Q.finite(x) & ~Q.zero(x) & ~Q.finite(y)) is False`

Yea, I made the changes.

**Review by TiloRC — CHANGES_REQUESTED** [review]

File: sympy/assumptions/tests/test_query.py [hunk 9]
```diff
@@ -1112,6 +1112,13 @@ def test_issue_27441():
     assert ask(Q.composite(y), Q.integer(y) & Q.positive(y) & ~Q.prime(y)) is None
 
 
+def test_issue_27447():
+    x,y = symbols('x y')
+    a = x * y
+    assert ask(Q.finite(a), Q.infinite(y)) is None
+    assert ask(Q.finite(a), Q.finite(x) & ~Q.zero(x) & ~Q.finite(y)) is False
+
+
```
Comment (Jatinbhardwaj-093) [inline sympy/assumptions/tests/test_query.py:1121]: Should I remove this test?

File: sympy/assumptions/tests/test_query.py [hunk 10]
```diff
@@ -1032,23 +1032,23 @@ def test_bounded():
...
-        & ~Q.finite(z)) is False
+    assert ask(Q.finite(a), Q.infinite(x) & Q.finite(y)
+        & Q.finite(z)) is None
+    assert ask(Q.finite(a), Q.infinite(x) & Q.finite(y)
+        & Q.infinite(z)) is None
     assert ask(Q.finite(a), ~Q.finite(x) & Q.finite(y)) is None
-    assert ask(Q.finite(a), ~Q.finite(x) & ~Q.finite(y)
-        & Q.finite(z)) is False
+    assert ask(Q.finite(a), Q.infinite(x) & Q.infinite(y)
+        & Q.finite(z)) is None
```
[the remark this answers is not in the document]
Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1051]: Oh sorry, I thought you suggested the change for recording in the new test which was added.

File: sympy/assumptions/tests/test_query.py (hunk 9 above)
Comment (TiloRC) [reply sympy/assumptions/tests/test_query.py:1121]: No. It's important to have this test. Maybe you should only include things that assert None though. It might be good to include all of the versions of the tests that assert None here.

Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py]: I’ve added the test case. If you have time, please take a look.

File: sympy/assumptions/handlers/calculus.py [hunk 11]
```diff
@@ -151,10 +151,12 @@ def _(expr, assumptions):
         * /s = not signed
     """
     result = True
+    possible_zero = False
     for arg in expr.args:
         _bounded = ask(Q.finite(arg), assumptions)
         if _bounded:
-            continue
+            if ask(Q.zero(arg), assumptions) is None:
+                possible_zero = None
```
Comment (TiloRC) [inline sympy/assumptions/handlers/calculus.py:159]: If result is False, you could immediately return None here. That would be more efficient.

File: sympy/assumptions/tests/test_query.py [hunk 12]
```diff
@@ -1112,6 +1112,27 @@ def test_issue_27441():
     assert ask(Q.composite(y), Q.integer(y) & Q.positive(y) & ~Q.prime(y)) is None
 
 
+def test_issue_27447():
+    x,y,z = symbols('x y z')
+    a = x*y
+    assert ask(Q.finite(a), Q.finite(x)  & ~Q.finite(y)) is None
+    assert ask(Q.finite(a), ~Q.finite(x)  & Q.finite(y)) is None
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:1119]: Could you add the following test? I believe it currently fails. 

`assert ask(Q.finite(a), ~Q.finite(x) & Q.zero(y)) is None`

File: sympy/assumptions/handlers/calculus.py [hunk 13]
```diff
@@ -164,7 +166,9 @@ def _(expr, assumptions):
                 result = None
         else:
             result = False
```
Comment (TiloRC) [inline sympy/assumptions/handlers/calculus.py:168]: If possible_zero is None, you can immediately return None here.

**Review by TiloRC — CHANGES_REQUESTED** [review]

File: sympy/assumptions/tests/test_query.py (hunk 12 above)
Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1119]: Thanks for finding it out!

I tried changing the if statement to `if ask(Q.zero(arg), assumptions) is not False:`, but I encountered an error related to the `ask` function. It seems to be due to the heuristic approach used by` ask`, and when it reaches the `satask` stage, it doesn’t handle the error well. So, I wrapped it in a try-except block.

https://github.com/sympy/sympy/blob/cc300e2d39576163eb17cfb10e6c1f65b1b002ea/sympy/assumptions/ask.py#L492-L515

File: sympy/assumptions/ask.py [hunk 14]
```diff
@@ -503,9 +503,12 @@ def ask(proposition, assumptions=True, context=global_assumptions):
...
     # using satask (still costly)
-    res = satask(proposition, assumptions=assumptions, context=context)
-    if res is not None:
-        return res
+    try:
+        res = satask(proposition, assumptions=assumptions, context=context)
+        if res is not None:
+            return res
+    except ValueError:
+        pass
```
Comment (TiloRC) [inline sympy/assumptions/ask.py:511]: You shouldn't be modifying this file. This try, except removes important functionality.

Comment (Jatinbhardwaj-093) [reply sympy/assumptions/ask.py:511]: I understand the concern. So issue needs to resolve in  `satask`, which further calls the `check_satisfiability` function.
https://github.com/sympy/sympy/blob/0fd9bdd788ab5f0b1ee1cc9580f6ef11d3153602/sympy/assumptions/satask.py#L85-L106

In this case, both can_be_false and can_be_true are False. I’m not sure if this is the expected behavior, but it seems likely that the missing `TODO` section should handle this scenario. However, I am uncertain about what the general case should be for the `TODO`.

Am I missing something, or should this issue indeed be resolved by implementing the missing `TODO` section?

File: sympy/assumptions/tests/test_query.py (hunk 12 above)
Comment (TiloRC) [reply sympy/assumptions/tests/test_query.py:1119]: This is caused by a bug: https://github.com/sympy/sympy/issues/27662.

For now can you making this an xfailing test? https://docs.sympy.org/latest/contributing/new-contributors-guide/writing-tests.html#writing-tests-xfail

File: sympy/assumptions/handlers/calculus.py [hunk 15]
```diff
@@ -163,8 +168,13 @@ def _(expr, assumptions):
             if result is not False:
                 result = None
         else:
-            result = False
-    return result
+            if possible_zero is None:
+                return None
+            else:
```
Comment (TiloRC) [inline sympy/assumptions/handlers/calculus.py:173]: no need for this else statement

Comment (TiloRC) [inline sympy/assumptions/handlers/calculus.py:173]: no need for this else statement

File: sympy/assumptions/handlers/calculus.py [hunk 16]
```diff
@@ -151,10 +151,15 @@ def _(expr, assumptions):
...
     result = True
+    possible_zero = False
     for arg in expr.args:
         _bounded = ask(Q.finite(arg), assumptions)
         if _bounded:
-            continue
+            if ask(Q.zero(arg), assumptions) is not False:
+                if result is False:
+                    return None
+                else:
```
Comment (TiloRC) [inline sympy/assumptions/handlers/calculus.py:161]: again — no need for else statement

File: sympy/assumptions/handlers/calculus.py [hunk 17]
```diff
@@ -163,8 +168,13 @@ def _(expr, assumptions):
...
         else:
-            result = False
-    return result
+            if possible_zero is None:
+                return None
+            else:
+                result = False
+    if result:
+        return result
+    return possible_zero
```
Comment (TiloRC) [inline sympy/assumptions/handlers/calculus.py:177]: I believe you can go back to just returning result here. If result is False and possible_zero is not False, then we would have already returned and not gotten to this point.

So the only situation where possible_zero would be returned would be when possible_zero is False and result is also False. So simply returning result would do the same thing.

**Review by TiloRC — CHANGES_REQUESTED** [review]

File: sympy/assumptions/tests/test_query.py (hunk 12 above)
Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1119]: Okay, I’ll try to solve it soon. For now, I’m placing this case in `test_bounded_xail`.

Comment (TiloRC) [reply sympy/assumptions/tests/test_query.py:1119]: Maybe add a comment next to the test referencing #27662

File: sympy/assumptions/tests/test_query.py [hunk 18]
```diff
@@ -1112,11 +1112,35 @@ def test_issue_27441():
...
+        & Q.finite(z)) is None
+
+
 @XFAIL
 def test_bounded_xfail():
     """We need to support relations in ask for this to work"""
     assert ask(Q.finite(sin(x)**x)) is True
     assert ask(Q.finite(cos(x)**x)) is True
+    assert ask(Q.finite(x*y), ~Q.finite(x)
+        & Q.zero(y)) is None
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:1143]: You should put this in a separate test as the reasons why this is failing is unrelated to the other xfailing tests here. Or actually maybe don't even add it at all.

Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py:1143]: Should I put the comment next to it or make a new function?

Comment (Jatinbhardwaj-093) [reply sympy/assumptions/tests/test_query.py]: I made the seperate test case by the name of `issue 27662`. Please take a look.

File: sympy/assumptions/tests/test_query.py [hunk 19]
```diff
@@ -1112,6 +1112,32 @@ def test_issue_27441():
...
+        & Q.finite(z) ) is None
+    assert ask(Q.finite(a), Q.finite(x) & ~Q.finite(y)
+        & ~Q.finite(z)) is None
+    assert ask(Q.finite(a), ~Q.finite(x) & Q.finite(y)
+        & Q.finite(z)) is None
+    assert ask(Q.finite(a), ~Q.finite(x) & Q.finite(y)
+        & ~Q.finite(z)) is None
+    assert ask(Q.finite(a), ~Q.finite(x) & ~Q.finite(y)
+        & Q.finite(z)) is None
+
```
Comment (TiloRC) [inline sympy/assumptions/tests/test_query.py:1134]: you're missing a space here

**Comment by TiloRC** [comment]
Your release note entry can be improved. Before this PR, if you were using SymPy and had used its assumption system before, would you have any idea what the release note entry meant?

File: sympy/assumptions/handlers/calculus.py [hunk 20]
```diff
@@ -151,10 +151,14 @@ def _(expr, assumptions):
...
     result = True
+    possible_zero = False
     for arg in expr.args:
         _bounded = ask(Q.finite(arg), assumptions)
         if _bounded:
-            continue
+            if ask(Q.zero(arg), assumptions) is not False:
+                if result is False:
+                    return None
+                possible_zero = None
```
Comment (TiloRC) [inline sympy/assumptions/handlers/calculus.py:161]: I think it would make more sense to just set possible_zero to True

**Comment by Jatinbhardwaj-093** [comment]
> Your release note entry can be improved. Before this PR, if you were using SymPy and had used its assumption system before, would you have any idea what the release note entry meant?

It's almost a year using sympy library but most of the time I use it only for calculus-related stuff.
For the release note, I think it is just the document that mention what features are added and bugs got patch in that version.
Can you suggest a better release note?

**Comment by Jatinbhardwaj-093** [comment]
Faling test related to pymc new release mention in [comment](https://github.com/sympy/sympy/pull/27664#discussion_r1971781385)

**Comment by TiloRC** [comment]
I suppose the release note doesn't really matter that much

> Can you suggest a better release note?

Maybe something like:

Make `ask(Q.finite(f * i))` give None instead of wrongly returning False (for finite f and infinite i)

**Comment by TiloRC** [comment]
> Can you suggest a better release note?

Actually, after rereading your release note entry, I actually like it better than mine.

**Comment by TiloRC** [comment]
Looks good. Thank you!

**Comment by Jatinbhardwaj-093** [comment]
Thank you @TiloRC for all the guidance and support throughout this PR.
